core/CMakeLists.txt

```diff
@@ -58,14 +58,24 @@ if(GINKGO_HAVE_PAPI_SDE)
     target_sources(ginkgo PRIVATE log/papi.cpp)
 endif()
 
+if (GINKGO_BUILD_MPI)
+    target_link_libraries(ginkgo PUBLIC MPI::MPI_CXX)
+    target_include_directories(ginkgo SYSTEM PUBLIC ${MPI_INCLUDE_PATH})
+endif()
+
 ginkgo_compile_features(ginkgo)
 
 target_compile_options(ginkgo PRIVATE "${GINKGO_COMPILER_FLAGS}")
 # add a namespace alias so Ginkgo can always be included as Ginkgo::ginkgo
 # regardless of whether it is installed or added as a subdirectory
 add_library(Ginkgo::ginkgo ALIAS ginkgo)
-target_link_libraries(ginkgo
-    PUBLIC ginkgo_device ginkgo_omp ginkgo_cuda ginkgo_reference ginkgo_hip ginkgo_dpcpp)
+if(GINKGO_BUILD_MPI)
+    target_link_libraries(ginkgo
+        PUBLIC ginkgo_device ginkgo_omp ginkgo_mpi ginkgo_cuda ginkgo_reference ginkgo_hip ginkgo_dpcpp)
+else()
+    target_link_libraries(ginkgo
+        PUBLIC ginkgo_device ginkgo_omp ginkgo_cuda ginkgo_reference ginkgo_hip ginkgo_dpcpp)
+endif()
```

Comment:
I think the idea was to have no MPI libraries/functions or anything if MPI is disabled. But I understand that now you have some asymmetry. it would be possible to add an empty library for MPI, but I am not sure it is necessary.

Reply:
I remove the entire mpi library now and just add the functions (currently only exception handling) into core